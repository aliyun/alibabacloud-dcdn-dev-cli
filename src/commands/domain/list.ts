import { CommandModule, ArgumentsCamelCase, Argv } from 'yargs';
import { checkDirectory, checkIsLoginSuccess } from '../utils.js';
import logger from '../../libs/logger.js';
import { getProjectConfig } from '../../utils/fileUtils/index.js';
import { GetRoutineReq, RelatedRecordProps } from '../../libs/interface.js';
import { ApiService } from '../../libs/apiService.js';
import t from '../../i18n/index.js';
import { validRoutine } from '../../utils/checkIsRoutineCreated.js';

const listDomain: CommandModule = {
  command: 'list',
  describe: `🔍 ${t('domain_list_describe').d('List all related domains')}`,
  handler: async (argv: ArgumentsCamelCase) => {
    handleListDomains(argv);
  }
};

export default listDomain;

export async function handleListDomains(argv: ArgumentsCamelCase) {
  if (!checkDirectory()) {
    return;
  }

  const projectConfig = getProjectConfig();
  if (!projectConfig) return logger.notInProject();

  const isSuccess = await checkIsLoginSuccess();
  if (!isSuccess) return;

  await validRoutine(projectConfig.name);

  const server = await ApiService.getInstance();

  const req: GetRoutineReq = { Name: projectConfig.name };
  const routineDetail = await server.getRoutine(req);

  if (!routineDetail) return;
  const relatedRecords: RelatedRecordProps[] =
    routineDetail.data?.RelatedRecords ?? [];

  if (relatedRecords.length === 0) {
    logger.log(`🙅 ${t('domain_list_empty').d('No related domains found')}`);
    return;
  }
  const domainList = relatedRecords.map((record) => record.RecordName);
  logger.log(`📃 ${t('domain_list_title').d('Related domains:')}`);
  logger.tree(domainList);
}
