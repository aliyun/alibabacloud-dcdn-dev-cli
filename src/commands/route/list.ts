import { CommandModule, ArgumentsCamelCase, Argv } from 'yargs';
import { getProjectConfig } from '../../utils/fileUtils/index.js';
import { GetRoutineReq, RelatedRouteProps } from '../../libs/interface.js';
import Table from 'cli-table3';
import { checkDirectory, checkIsLoginSuccess } from '../utils.js';
import { ApiService } from '../../libs/apiService.js';
import logger from '../../libs/logger.js';
import t from '../../i18n/index.js';
import { validRoutine } from '../../utils/checkIsRoutineCreated.js';

const listRoute: CommandModule = {
  command: 'list',
  describe: `🔍 ${t('route_list_describe').d('List all related routes')}`,
  handler: async () => {
    handleListRoutes();
  }
};

export default listRoute;

export async function handleListRoutes() {
  if (!checkDirectory()) {
    return;
  }

  const projectConfig = getProjectConfig();
  if (!projectConfig) return logger.notInProject();
  const isSuccess = await checkIsLoginSuccess();
  if (!isSuccess) return;

  await validRoutine(projectConfig.name);

  const server = await ApiService.getInstance();
  const req: GetRoutineReq = { Name: projectConfig.name };

  const routineDetail = await server.getRoutine(req);
  if (!routineDetail) return;
  const relatedRoutes: RelatedRouteProps[] =
    routineDetail.data?.RelatedRoutes ?? [];

  if (relatedRoutes.length === 0) {
    logger.warn(`🙅 ${t('route_list_empty').d('No related routes found')}`);
    return;
  }
  logger.log(`📃 ${t('route_list_title').d('Related routes')}:`);
  displayRelatedRouteList(relatedRoutes);
}

export async function displayRelatedRouteList(routeList: RelatedRouteProps[]) {
  const table = new Table({
    head: ['Route', 'Site'],
    colWidths: [30, 30]
  });
  for (let i = 0; i < routeList.length; i++) {
    const route = routeList[i];
    table.push([route.Route, route.SiteName]);
  }
  console.log(table.toString());
}
