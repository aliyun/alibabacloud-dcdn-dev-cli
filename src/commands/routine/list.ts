import { CommandModule, ArgumentsCamelCase, Argv } from 'yargs';
import { EdgeFunctionItem, ListSitesReq } from '../../libs/interface.js';
import Table from 'cli-table3';
import logger from '../../libs/logger.js';
import { Base64 } from 'js-base64';
import { checkIsLoginSuccess } from '../utils.js';
import chalk from 'chalk';
import { ApiService } from '../../libs/apiService.js';
import t from '../../i18n/index.js';

const list: CommandModule = {
  command: 'list',
  describe: `ðŸ“‹ ${t('list_describe').d('List all your routines')}`,
  builder: (yargs: Argv) => {
    return (
      yargs
        // .positional('deployment', {
        //   type: 'string',
        //   description: t('list_deployment_positional_describe').d(
        //     '(Optional) ID or URL of the deployment to tail. Specify by environment if deployment ID is unknown.'
        //   ),
        //   demandOption: true
        // })
        // .option('site', {
        //   alias: 's',
        //   describe: t('list_site_option_describe').d(
        //     'List all site names of your account'
        //   ),
        //   type: 'boolean',
        //   default: false
        // })
        .usage(`${t('common_usage').d('Usage')}: \$0 list []`)
    );
  },
  handler: async (argv: ArgumentsCamelCase) => {
    handleList(argv);
  }
};

export default list;

export async function handleList(argv: ArgumentsCamelCase) {
  const { site, ...args } = argv;
  const isSuccess = await checkIsLoginSuccess();
  if (!isSuccess) return;
  const server = await ApiService.getInstance();

  if (site) {
    const req: ListSitesReq = {
      SiteSearchType: 'fuzzy',
      Status: 'active',
      PageNumber: 1,
      PageSize: 50
    };
    const res = await server.listSites(req);
    const siteList = res?.data.Sites ?? [];
    const siteNameList: string[] = siteList?.map((item: any) => item.SiteName);
    logger.log(
      chalk.bold.bgGray(
        `ðŸ“ƒ ${t('list_site_name_title').d('List all of site names')}:`
      )
    );
    logger.tree(siteNameList);
    return;
  }

  const res = await server.getRoutineUserInfo();
  const routineList = res?.Routines;
  if (routineList) {
    console.log(
      chalk.bold.bgGray(
        `ðŸ“ƒ ${t('list_routine_name_title').d('List all of routine')}:`
      )
    );
    displayRoutineList(routineList);
  }
}

export async function displayRoutineList(versionList: EdgeFunctionItem[]) {
  const table = new Table({
    head: ['Name', 'Created', 'Description', 'Specification']
  });

  versionList.forEach((version) => {
    table.push([
      version.RoutineName,
      new Date(version.CreateTime).toLocaleString(),
      Base64.decode(version.Description),
      version.SpecName ?? ' '
    ]);
  });

  console.log(table.toString());
}
