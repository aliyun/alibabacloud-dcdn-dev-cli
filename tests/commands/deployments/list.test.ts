import { it, describe, expect, vi } from 'vitest';
import { handleListDeployments } from '../../../src/commands/deployments/list.js';
import { mockConsoleMethods } from '../../helper/mockConsole.js';
import { ApiService } from '../../../src/libs/apiService.js';

describe('handle display deployments', () => {
  let std = mockConsoleMethods();

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should handle display deployments success', async () => {
    await handleListDeployments({
      _: [],
      $0: ''
    });
    expect(std.out).matchSnapshot();
  });

  it('should handle display deployments -- show env ip error', async () => {
    vi.mocked(
      (await ApiService.getInstance()).getRoutineStagingEnvIp
    ).mockResolvedValue({} as any);
    await handleListDeployments({
      _: [],
      $0: ''
    });
    expect(std.out).matchSnapshot();
  });
});
