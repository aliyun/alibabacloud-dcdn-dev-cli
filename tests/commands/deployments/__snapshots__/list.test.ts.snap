// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handle display deployments > should handle display deployments -- show env ip error 1`] = `
[MockFunction log] {
  "calls": [
    [
      "[1mStaging[22m",
    ],
    [
      "[90m┌───────────────[39m[90m┬───────────────────┐[39m
[90m│[39m Version       [90m│[39m productionVersion [90m│[39m
[90m├───────────────[39m[90m┼───────────────────┤[39m
[90m│[39m Specification [90m│[39m 100ms             [90m│[39m
[90m└───────────────[39m[90m┴───────────────────┘[39m",
    ],
    [
      " ",
    ],
    [
      " ",
    ],
    [
      "[1mProduction [32m●[39m[22m",
    ],
    [
      "[90m┌───────────────[39m[90m┬────────────────┐[39m
[90m│[39m Version       [90m│[39m stagingVersion [90m│[39m
[90m├───────────────[39m[90m┼────────────────┤[39m
[90m│[39m Specification [90m│[39m 50ms           [90m│[39m
[90m└───────────────[39m[90m┴────────────────┘[39m",
    ],
    [
      " ",
    ],
    [
      " ",
    ],
    [
      "[1mCanary [22m",
    ],
    [
      "[90m┌───────────────[39m[90m┬──────┐[39m
[90m│[39m Version       [90m│[39m      [90m│[39m
[90m├───────────────[39m[90m┼──────┤[39m
[90m│[39m Specification [90m│[39m 50ms [90m│[39m
[90m└───────────────[39m[90m┴──────┘[39m",
    ],
    [
      " ",
    ],
    [
      "💬 You can visit: [93mundefined[39m",
    ],
    [
      " ",
    ],
    [
      "[43mActive[49m Staging",
    ],
    [
      "[42mActive[49m Production",
    ],
    [
      "[90m┌────────────[39m[90m┬───────────────────[39m[90m┬─────────────┐[39m
[90m│[39m[31m Version    [39m[90m│[39m[31m Created           [39m[90m│[39m[31m Description [39m[90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m unstable   [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m v1         [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m v2         [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m└────────────[39m[90m┴───────────────────[39m[90m┴─────────────┘[39m",
    ],
    [
      " ",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`handle display deployments > should handle display deployments success 1`] = `
[MockFunction log] {
  "calls": [
    [
      "[1mStaging[22m",
    ],
    [
      "Staging IP: [32m0.0.0.0[39m",
    ],
    [
      "[90m┌───────────────[39m[90m┬───────────────────┐[39m
[90m│[39m Version       [90m│[39m productionVersion [90m│[39m
[90m├───────────────[39m[90m┼───────────────────┤[39m
[90m│[39m Specification [90m│[39m 100ms             [90m│[39m
[90m└───────────────[39m[90m┴───────────────────┘[39m",
    ],
    [
      " ",
    ],
    [
      " ",
    ],
    [
      "[1mProduction [32m●[39m[22m",
    ],
    [
      "[90m┌───────────────[39m[90m┬────────────────┐[39m
[90m│[39m Version       [90m│[39m stagingVersion [90m│[39m
[90m├───────────────[39m[90m┼────────────────┤[39m
[90m│[39m Specification [90m│[39m 50ms           [90m│[39m
[90m└───────────────[39m[90m┴────────────────┘[39m",
    ],
    [
      " ",
    ],
    [
      " ",
    ],
    [
      "[1mCanary [22m",
    ],
    [
      "[90m┌───────────────[39m[90m┬──────┐[39m
[90m│[39m Version       [90m│[39m      [90m│[39m
[90m├───────────────[39m[90m┼──────┤[39m
[90m│[39m Specification [90m│[39m 50ms [90m│[39m
[90m└───────────────[39m[90m┴──────┘[39m",
    ],
    [
      " ",
    ],
    [
      "💬 You can visit: [93mundefined[39m",
    ],
    [
      " ",
    ],
    [
      "[43mActive[49m Staging",
    ],
    [
      "[42mActive[49m Production",
    ],
    [
      "[90m┌────────────[39m[90m┬───────────────────[39m[90m┬─────────────┐[39m
[90m│[39m[31m Version    [39m[90m│[39m[31m Created           [39m[90m│[39m[31m Description [39m[90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m unstable   [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m v1         [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m├────────────[39m[90m┼───────────────────[39m[90m┼─────────────┤[39m
[90m│[39m v2         [90m│[39m 2021/1/1 08:00:00 [90m│[39m             [90m│[39m
[90m└────────────[39m[90m┴───────────────────[39m[90m┴─────────────┘[39m",
    ],
    [
      " ",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
